enhancements/proposals/2490-scylladbmonitoring-external-prometheus/README.md

```diff
@@ -0,0 +1,611 @@
+   type LocalObjectKeySelector struct {
+      // name of the selected object.
+      // +kubebuilder:validation:MinLength=1
+      Name string `json:"name"`
+
+      // key within the selected object.
+      // +kubebuilder:validation:MinLength=1
+      Key string `json:"key"`
+   }
+
+   type GrafanaDatasourceTLSSpec struct {
+      // caCert is a reference to a key within the CA bundle ConfigMap. The key should hold the CA cert in PEM format.
+      // When not specified, system CAs are used.
+      // +optional
+      CACertConfigMapRef *LocalObjectKeySelector `json:"caCertConfigMapRef,omitempty"`
+
+      // insecureSkipVerify controls whether to skip server certificate verification.
+      // +kubebuilder:default:=false
+      // +optional
+      InsecureSkipVerify bool `json:"insecureSkipVerify,omitempty"`
+
+	  // clientTLSKeyPairSecretRef is a name of a Secret holding client TLS certificate and key for mTLS authentication.
+	  // It's expected to be a standard Kubernetes TLS Secret with `tls.crt` and `tls.key` keys.
+      // +optional
+      ClientTLSKeyPairSecretName *corev1.LocalObjectReference `json:"clientTLSKeyPairSecretRef,omitempty"`
+   }
+   ```
+
+   `GrafanaPrometheusDatasourceAuthSpec` will define authentication options, starting with bearer token auth. It will
+   be a separate struct to allow adding more auth methods in the future if needed (e.g., basic auth).
+
+   ```go
+   type GrafanaPrometheusDatasourceAuthType string
+   
+   const (
+      // GrafanaPrometheusDatasourceAuthTypeNone means no authentication.
+      GrafanaPrometheusDatasourceAuthTypeNone GrafanaPrometheusDatasourceAuthType = "None"
+
+      // GrafanaPrometheusDatasourceAuthTypeBearerToken means Bearer token authentication.
+      GrafanaPrometheusDatasourceAuthTypeBearerToken GrafanaPrometheusDatasourceAuthType = "BearerToken"
+    )
+   
+   type GrafanaPrometheusDatasourceAuthSpec struct {
+	  // type is the type of authentication to use.
+      // +kubebuilder:default:="None"
+      // +optional
+      Type GrafanaPrometheusDatasourceAuthType `json:"type,omitempty"`
+   
+      // bearerToken holds options for Bearer token authentication.
+      // +optional
+      BearerTokenOptions *GrafanaPrometheusDatasourceBearerTokenAuthSpec `json:"bearerTokenOptions,omitempty"`
+   }
+
+   type GrafanaPrometheusDatasourceBearerTokenAuthOptions struct {
+      // secretRef is a reference to a key in a Secret holding a Bearer token to use to authenticate with Prometheus.
+      // +optional
+      SecretRef *LocalObjectKeySelector `json:"secretRef,omitempty"`
+   }
+   ```
+
+On every change to the `datasources` field, the Grafana deployment will be rerolled to apply the new
+configuration. It will be achieved by adding an annotation to the Grafana deployment's Pod template with a hash of the
+data source configuration and referenced `Secret`/`ConfigMap` names and keys.
```

Comment:
https://github.com/scylladb/scylla-operator/blob/6d673e2aaba4070a21460f8534cf69d632d67508/pkg/controller/scylladbmonitoring/sync_grafana.go#L338-L339

We already have this functionality. In case of change in Grafana datasource, provisioning CM will change which is part of inputs hash of grafana deployment.
https://github.com/scylladb/scylla-operator/blob/9d36ac3d636fd29124ad9ebd2086829a43f2f3a7/assets/monitoring/grafana/v1alpha1/deployment.yaml#L14

Reply:
Reworded this paragraph to make that clear.